libs/blockscout-service-launcher/src/test_server/server.rs

```diff
@@ -0,0 +1,90 @@
+use crate::launcher::ServerSettings;
+use reqwest::Url;
+use std::{
+    future::Future,
+    net::{SocketAddr, TcpListener},
+    str::FromStr,
+    time::Duration,
+};
+use tokio::time::timeout;
+
+fn get_free_port() -> u16 {
+    let listener = TcpListener::bind("127.0.0.1:0").unwrap();
+    listener.local_addr().unwrap().port()
+}
+
+pub fn get_test_server_settings() -> (ServerSettings, Url) {
+    let mut server = ServerSettings::default();
+    let port = get_free_port();
+    server.http.addr = SocketAddr::from_str(&format!("127.0.0.1:{port}")).unwrap();
+    server.grpc.enabled = false;
+    let base = Url::parse(&format!("http://{}", server.http.addr)).unwrap();
+    (server, base)
+}
+
+pub async fn init_server<F, R>(run: F, base: &Url, health_check_service: &str)
+where
+    F: FnOnce() -> R + Send + 'static,
+    R: Future<Output = ()> + Send,
+{
+    tokio::spawn(async move { run().await });
+
+    let client = reqwest::Client::new();
+    let health_endpoint = base.join("health").unwrap();
+
+    let wait_health_check = async {
+        loop {
+            if let Ok(_response) = client
+                .get(health_endpoint.clone())
+                .query(&[("service", health_check_service)])
```

Comment:
I'm not sure, but the actual service name probably may be omitted. If I'm not mistaken, the most important is that the request should contain `service=` query inside. But it may be left as it is without any concrete service names. We would ignore the name anyway

Reply:
Fixed